### PR TITLE
Initial source code

### PR DESCRIPTION
**Pull request by inteon**
The initial source code was created and battle-tested by Jetstack/ Venafi based on one of their enterprise issuers.
Jetstack/Venafi proposed to donate this source code to the cert-manager organisation.
We as cert-manager maintainers think that this project can be of great value to our community and will help us write new issuers and improve existing issuers.

The long-term goal is to use this library as a base for all external issuers and even as a base for the core issuers.
Short-term, we aim to improve this library and make developing new external issuers much easier.

See https://cert-manager.io/docs/contributing/third-party-code-donation/ for more info on the process.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,8 +14,15 @@ See `./internal/testsetups/simple` for an example of creating an issuer based on
 
 ---
 
-⚠️ Warning: This library is ready to be used in production, but the API is still subject to change.
-Developers using this library might have to update their code when updating to a newer version.
+⚠️ Warning: This library's API is still subject to change. Developers using this library will have to update their
+code when updating to a newer version.
+
+0. Currently, this library is used to build production Issuers, but no open-source Issuers/ examples are available yet. We advice to use this library only for experimentation.
```
Comment (irbekrm) [inline README.md:20]: Nit:

```suggestion
0. Currently, this library is used to build production Issuers, but no open-source Issuers/ examples are available yet. We advise to use this library only for experimentation.
1. ```

**Review by irbekrm — APPROVED** [review]
I've just had a quick read through the code as I think we semi-agreed in the previous standups that we aren't going to do an in-depth code review as this already has been reviewed and there is too much to review for one commit.

Thank you for updating the Readme, I think it is important that we communicate clearly the production readiness. At the same time I agree that to actually make it useful more eyes on it would be good, perhaps you want to talk about how this can be used in the next dev meeting or use some other means to invite people who are willing to play around with a less stable issuer implementation.

Adding a hold in case others still want to review, but feel free to unhold when ready

/hold
/lgtm

**Review by SgtCoDFish — APPROVED** [review]
/lgtm
/unhold

I think we should merge this. The idea is sound, we know how we'll test this and graduate it, we've got warnings reflecting the level of support and how alpha this is. I can't think of a good reason to delay it!

**Comment by SgtCoDFish** [comment]
Oh, the missing OWNERS file might be the next step. I'/ll add the approved label manually for now!

**Pull request merged**